src/registrar/admin.py

```diff
@@ -716,16 +719,58 @@ def response_change(self, request, obj):
         return super().response_change(request, obj)
 
     def do_delete_domain(self, request, obj):
+        if not isinstance(obj, Domain):
+            # Could be problematic if the type is similar,
+            # but not the same (same field/func names) so we err out.
+            # We do not want to accidentally delete records.
+            raise ValueError("Object is not of type Domain")
         try:
-            obj.deleted()
+            obj.deletedInEpp()
             obj.save()
-        except Exception as err:
-            self.message_user(request, err, messages.ERROR)
+        except RegistryError as err:
+            # To get past the linter..
```

Comment:
remove .. at end of comment. and I would change this to Using variables to get past linter. As it is not clear what is being done here linter wise